varinfo/cmr_search.py

```diff
@@ -134,3 +143,41 @@ def download_granule(granule_link: str,
         # Custom exception for error from `requests.get`
         raise GranuleDownloadException(
             str(requests_exception)) from requests_exception
+
+
+def get_edl_token_from_launchpad(launchpad_token: str,
+                                 cmr_env: CmrEnvType) -> Union[str, None]:
+    ''' Retrieve an EDL token given a LaunchPad token.
+        * launchpad_token: A LaunchPad token with no header prefixes:
+            <Launchpad token>
+        * cmr_env/mode: CMR environments (OPS, UAT, and SIT)
+    '''
+    url_urs_endpoint = UrsEnvsEdlTokenEndpoints.get(cmr_env)
+    try:
+        response = requests.post(url=url_urs_endpoint,
+                                 data=f'token={launchpad_token}',
+                                 timeout=10)
+        response.raise_for_status()
+
+    except Exception as requests_exception:
```

Comment:
I do see your point with this and if you feel strongly about it I can change this. My thinking is having this generic `Exception`, like we have above with `download_granules`, made the most sense because it covers all of the potential exceptions ('HTTP', TimeOut, etc), that `request` may have. And when we `raise GetEdlTokenException` we can see the specific `request.exception` that was raised, but I don't know if we want `pycodestyle` to complain lol.

Reply:
I thought that `requests` would likely allow us to specify one or two high level exception classes (of which there are probably many children), but I am fine to leave this as-is.